viewer/core/static/js/app/actions/url-actions.js

```diff
@@ -0,0 +1,17 @@
+import * as types from './action-types';
+
+
+export function addMissingQueryParameters(currentQueryParameters) {
+  return {
+    type: types.ADD_MISSING_QUERY_PARAMETERS,
+    currentQueryParameters,
+  };
+}
+
+export function updateQueryParameter(key, newValue) {
+  return {
+    type: types.UPDATE_QUERY_PARAMETER,
+    key,
+    newValue,
+  };
+}
```

Comment:
I haven't seen the rest and *this* I love! Exactly what I was hoping we'd get to.